### PR TITLE
Add external load balancer template

### PR DESCRIPTION
**Pull request by sourced-pavel**

### CODE REVIEW

**Comment by JvetS** [comment]
URL map reviewed.
Continuing with other sections.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+imports:
+  - path: templates/ssl_cerficate/ssl_cerficate.py
+    name: ssl_certificate.py
+
+resources:
+  - name: example-certificate
+    type: ssl_certificate.py
+    properties:
+      certificate: |
+        -----BEGIN CERTIFICATE-----
```
Comment (JvetS) [inline community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Should be `<FIXME:certificateContents>`.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        SIcc4YIlKOB0nFDKPK+iy7LqPwIDAQABMA0GCSqGSIb3DQEBBQUAA4IBAQAdf4dG
+        IkEmdNmTeGPVOUis+7ziWzRPxt8Vpmuq24z4H9mIkAPo/2aLpMKH7bloeYvz8blu
+        5VQZx7StoE5Sde1ss/AoaL1dVJi/dgmeN2cHy7J6POu3e9n6yXGiIh0qHlFe83nJ
+        RVIqtN9QGGuabGt3WGbKElMKwrCl9NGhExi/LntPFllXfTLb2pVGXH47ZihynbUj
+        4S21+KnQUPjhg6Na6hP3qLVqSYtWataJFpy6DOG1wgoAWjagNc3ltGdmv6O/ZkI3
+        3vymENyn8G7n+Z1knXUXxv4rJoeiYbZ7/2/bQ8BTc/RI5qnhzO8VYmyAZrrKGZnD
+        W2xuikK4nQHsideP
+        -----END CERTIFICATE-----
+      privateKey: |
+        -----BEGIN RSA PRIVATE KEY-----
```
Comment (JvetS) [inline community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Should be `<FIXME:privateKeyContents>`.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Example of the SSL certificate template usage.
+
```
Comment (JvetS) [inline community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Add:

    #
    # Replace the following placeholders with valid values:
    #   <FIXME:certificateContents>: contents of the certificate file
    #   <FIXME:privateKeyContents>: contents of the private key file

**Comment by JvetS** [comment]
The target_proxy integration test is missing a TCP proxy test.

File: community/cloud-foundation/templates/external_load_balancer/examples/external_load_balancer_tcp.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    name: external_load_balancer.py
+
+resources:
+  - name: example-tcp-elb
+    type: external_load_balancer.py
+    properties:
+      portRange: 80
+      backendServices:
+        - name: backend-service
+          portName: https
```
Comment (JvetS) [inline community/cloud-foundation/templates/external_load_balancer/examples/external_load_balancer_tcp.yaml]: Should be `http`.

**Comment by JvetS** [comment]
The external_load_balancer integration test is missing a TCP load balancer test.

**Comment by JvetS** [comment]
The integration tests contain hardcoded keys and certificates, but at least the certificate should be generated dynamically to avoid certificate expiration issues.

**Comment by JvetS** [comment]
Review concluded for this PR. I had a discussion with Adam about some of the items and concluded that the missing tests and dynamic certificates won't need to be fixed to accept the PR. Instead, we'll file issues and fix these separately.

File: community/cloud-foundation/templates/url_map/examples/url_map.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+imports:
+  - path: templates/url_map/url_map.py
+    name: url_map.py
+
+resources:
+  - name: test-url-map
+    type: url_map.py
+    properties:
+      urlMap:
```
[the remark this answers is not in the document]
Comment (sourced-vince) [reply community/cloud-foundation/templates/url_map/examples/url_map.yaml]: Fixed.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml (hunk 2 above)
Comment (sourced-vince) [reply community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Fixed.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml (hunk 1 above)
Comment (sourced-vince) [reply community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Fixed.

File: community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml (hunk 3 above)
Comment (sourced-vince) [reply community/cloud-foundation/templates/ssl_certificate/examples/ssl_certificate.yaml]: Fixed.

File: community/cloud-foundation/templates/external_load_balancer/examples/external_load_balancer_tcp.yaml (hunk 4 above)
Comment (sourced-vince) [reply community/cloud-foundation/templates/external_load_balancer/examples/external_load_balancer_tcp.yaml]: Fixed.

**Pull request merged**